code/modules/client/preferences/middleware/keybindings.dm

```diff
@@ -88,6 +88,8 @@
 		keybindings[keybinding.category][keybinding.name] = list(
 			"name" = keybinding.full_name,
 			"description" = keybinding.description,
+			"pref_key" = keybinding.required_pref_key,
```

Comment:
why does the UI need this data?

Reply:
To hide the keybindings if a certain preference is not a specific value, in this case we want to hide settings as the different modes have different hotkeys that can't be shared between them due to being fundamentally different and it would add a load of preferences that don't do anything to the menu making it harder to see which keybinds are effective.